crates/net/discv4/src/lib.rs

```diff
@@ -1387,7 +1387,22 @@ impl Discv4Service {
                 BucketEntry::SelfEntry => {
                     // we received our own node entry
                 }
-                _ => self.find_node(&closest, ctx.clone()),
+                _ => {
+                    let key = kad_key(closest.id);
```

Comment:
we already have `key` defined just before the match statement, and we also already do a lookup in `self.kbuckets` here, so we can make this two match arms instead

Reply:
totally missed it thx! done here https://github.com/paradigmxyz/reth/pull/8002/commits/b2a0b997ec90e14d6d00440047e631c2008007c7